libsolidity/codegen/ir/Common.cpp

```diff
@@ -127,15 +131,15 @@ string IRNames::zeroValue(Type const& _type, string const& _variableName)
 	return "zero_" + _type.identifier() + _variableName;
 }
 
-string sourceLocationComment(langutil::SourceLocation const& _location, IRGenerationContext const& _context)
+string sourceLocationComment(SourceLocation const& _location, IRGenerationContext const& _context)
 {
 	solAssert(_location.sourceName, "");
-	return "/// @src "
-		+ to_string(_context.sourceIndices().at(*_location.sourceName))
-		+ ":"
-		+ to_string(_location.start)
-		+ ":"
-		+ to_string(_location.end);
+	return AsmPrinter::formatSourceLocationComment(
+		_location,
+		_context.sourceIndices(),
+		true /* _statement */,
+		_context.soliditySourceProvider()
+	);
 }
```

Comment:
This is a piece of code duplicated from `AsmParser`. Because of this, my changes in `AsmParser` added snippets for `--ir-optimized` but not for `--ir`.

I moved this from both places into a generic static method in `AsmParser`.